packages/core/src/types.ts

```diff
@@ -186,7 +186,7 @@ export type CellStateStyle = {
    *
    * See {@link startArrow}.
    */
-  endArrow?: ArrowValue | string;
+  endArrow?: ArrowValue;
```

Comment:
**Note:** this remark applies to all property types changed in this Pull Request

It is not currently possible to remove the string type from the union.
As indicated in the property's JSDoc, the property must be able to accept custom elements registered by library users. These are registered under names that are not known to `maxGraph`. See #163 for details.

In addition, the changes proposed here break the tests and the ts-example package for this very reason. For instance, see https://github.com/maxGraph/maxGraph/actions/runs/6556791592/job/17862003925?pr=256.

There are maybe better ways to model this. If you are interested in this topic, I suggest you to open a discussion in https://github.com/maxGraph/maxGraph/discussions/categories/ideas to let us find a better solution together.

Regarding the guidance provided by the IDEs, I haven't seen any problem with IntelliJ/Webstrom, nor VSCode (and any Cloud Development Environment like Gitpod or GitHub Codespaces).
They should propose the known maxGraph values (for the current property, all strings of the ArrowValue union) and accept any strings.